providers/terraform-cloud-scanner/src/scanner.ts

```diff
@@ -0,0 +1,165 @@
+import { env } from "./config.js";
+import { api } from "./sdk.js";
+
+const workspaceTemplate = handlebars.compile(
+  env.CTRLPLANE_WORKSPACE_TARGET_NAME,
+);
+
+/**
+ * Scans Terraform Cloud workspaces and registers them as targets with prefixed labels and a link.
+ */
+export async function scan() {
+  logger.info("Starting Terraform Cloud scan");
+
+  try {
+    const workspaces: Workspace[] = await listWorkspaces();
+    logger.info(`Found ${workspaces.length} workspaces`);
+
+    const targets: SetTargetProvidersTargetsRequestTargetsInner[] = [];
+
+    for (const workspace of workspaces) {
+      logger.info(
+        `Processing workspace: ${workspace.attributes.name} (ID: ${workspace.id})`,
+      );
+
+      const variables: Variable[] = await listVariables(workspace.id);
+      logger.info(
+        `  Found ${variables.length} variables in workspace '${workspace.attributes.name}'`,
+      );
+      const variableLabels = processVariables(variables);
+      const tagLabels = processWorkspaceTags(workspace.attributes["tag-names"]);
+      const vcsRepoLabels = processVcsRepo(workspace.attributes["vcs-repo"]);
+      const link = buildWorkspaceLink(workspace);
+      const targetName = workspaceTemplate({ workspace });
+
+      const target: SetTargetProvidersTargetsRequestTargetsInner = {
+        version: "terraform/v1",
+        kind: "Workspace",
+        name: targetName,
+        identifier: workspace.id,
+        config: {
+          workspaceId: workspace.id,
+        },
+        labels: {
+          "terraform/organization": env.TFE_ORGANIZATION,
+          "terraform/workspace-name": workspace.attributes.name,
+          ...variableLabels,
+          ...tagLabels,
+          ...vcsRepoLabels,
+          "ctrlplane/link": link,
+        },
+      };
+
+      targets.push(target);
+    }
+
+    const uniqueTargets = _.uniqBy(targets, (t) => t.identifier);
+
+    logger.info(`Registering ${uniqueTargets.length} unique targets`);
+    const providerId = await getOrCreateProviderId();
+
+    if (!providerId) {
+      logger.error(
+        "Provider ID is not available. Aborting target registration.",
+      );
+      return;
+    }
+
+    await api.setTargetProvidersTargets({
+      providerId,
+      setTargetProvidersTargetsRequest: {
+        targets: uniqueTargets,
+      },
+    });
+
+    logger.info("Successfully registered targets");
+  } catch (error) {
+    logger.error("An error occurred during the scan process:", error);
+  }
+}
+
+/**
+ * Processes variables into Terraform variables and environment variables.
+ * @param variables The array of variables to process.
+ * @returns An object containing terraformVariables and envVariables.
+ */
+const processVariables = (variables: Variable[]) =>
+  Object.fromEntries(
+    variables.map((variable) => {
+      return [
+        variable.attributes.category === "terraform"
+          ? `var/${variable.attributes.key}`
+          : `env/${variable.attributes.key}`,
```

Comment:
whats the benefit to spliting this?

Reply:
It is the way that TF splits them out: 
<img width="1071" alt="Screen Shot 2024-09-16 at 10 21 19 PM" src="https://github.com/user-attachments/assets/fd963e01-9ec2-41e2-adfa-92e1b094f03c">

Reply:
There are terraform variables, and environment runtime variables.

Reply:
but they are both variables?

Reply:
They server completely different purposes. 
https://developer.hashicorp.com/terraform/cloud-docs/workspaces/variables 

If you want them to just be the same in our app we can, but I think that might lead to confusion when this is a PaaS.